packages/validator/src/util/params.ts

```diff
@@ -29,3 +65,131 @@ export function assertEqualParams(localConfig: IChainConfig, externalSpecJson: R
     throw new NotEqualParamsError("Not equal BeaconParams\n" + errors.join("\n"));
   }
 }
+
+function getSpecCriticalParams(localConfig: IChainConfig): Record<keyof ConfigWithPreset, boolean> {
+  const altairForkRelevant = localConfig.ALTAIR_FORK_EPOCH < Infinity;
+  const bellatrixForkRelevant = localConfig.BELLATRIX_FORK_EPOCH < Infinity;
+  const shardingForkRelevant = localConfig.SHARDING_FORK_EPOCH < Infinity;
+
+  return {
+    // # Config
+    ///////////
+
+    PRESET_BASE: false, // Not relevant, each preset value is checked below
+    TERMINAL_TOTAL_DIFFICULTY: bellatrixForkRelevant,
+    TERMINAL_BLOCK_HASH: bellatrixForkRelevant,
+    TERMINAL_BLOCK_HASH_ACTIVATION_EPOCH: bellatrixForkRelevant,
+
+    // Genesis
+    MIN_GENESIS_ACTIVE_VALIDATOR_COUNT: true,
```

Comment:
Instead of having a map of params to boolean, is it possible to have a list instead? Thus if a param is contained in the list, then it should be considered critical and be checked. 

I think with that, it becomes easier know the defined list and see what params are currently supported, instead of checking the boolean value in a map

Reply:
If we had a list of keys we lose the guarantee that Typescript provides that all keys have been manually assigned a "true" or "false" by a developer. Otherwise new keys will be assumed as "false" which is dangerous and not maintainable, as we would forget to add new keys.